sycl/doc/extensions/proposed/sycl_ext_oneapi_async_memory_alloc.asciidoc

```diff
@@ -0,0 +1,729 @@
+ memory as it is allocated from the OS to the memory pool will be initialised to
+ zero. Note there is no guarantee that the memory allocation be re-initialized
+ to zero when it is re-allocated from the pool, so users must re-initialize
+ memory to zero if they wish for later allocations to have this behaviour.
+
+|===
+
+
+=== Default memory pools
+
+As well as being able to construct a memory pool explicitly, this extension
+introduces a default memory pool per device for each SYCL context and device
+pair for device allocations and a default memory pool per context for host
+allocations.
+
+New member functions are added to the `context` class to retrieve the default
+memory pool as a copy of the `memory_pool` object. This can be modified and have
+those modifications reflected as it conforms to the SYCL common reference
+semantics.
+
+[source,c++]
+----
+class context {
+
+  memory_pool context::ext_oneapi_get_default_memory_pool();
+
+  memory_pool context::ext_oneapi_get_default_memory_pool(device dev);
+
+}; // context
+----
+
+[source, c++]
+----
+memory_pool context::ext_oneapi_get_default_memory_pool();
+----
+
+_Returns_: The default memory pool associated with the context for allocating
+with the allocation kind `usm::host`.
+
+[source, c++]
+----
+memory_pool context::ext_oneapi_get_default_memory_pool(device dev);
+----
+
+_Returns_: The default memory pool associated with the context and `dev` for
+allocating with the allocation kind `usm::device`.
+
+
+=== Asynchronous malloc & free
+
+This extension introduces a series of new enqueue functions for enqueueing
+asynchornous malloc and free commands which operate with the memory pools also
+introduced in this extension.
+
+All enqueue functions introduced have overloads which take a SYCL `queue` and a
+SYCL `handler`. None of enqueue functions return a SYCL `event` direction, as
+this extension is in line with the
+link:../experimental/sycl_ext_oneapi_enqueue_functions.asciidoc[
+  sycl_ext_oneapi_enqueue_functions] extension, so events are returned when
+calling `submit_with_event` and the `handler` overloads of these enqueue
+functions.
+
+[source,c++]
+----
+namespace ext::oneapi::experimental {
+
+void *async_malloc(queue q, size_t size);
+
+void *async_malloc(handler h, size_t size);
+
+void *async_malloc_from_pool(queue q, size_t size, memory_pool pool);
+
+void *async_malloc_from_pool(handler h, size_t size, memory_pool pool);
+
+void async_free(queue q, void *ptr);
+
+void async_free(handler h, void *ptr);
+
+}  // ext::oneapi::experimental
+----
+
+[source, c++]
+----
+void *async_malloc(queue q, size_t size);
+
+void *async_malloc(handler h, size_t size);
+
+void *async_malloc_from_pool(queue q, size_t size, memory_pool pool);
+
+void *async_malloc_from_pool(handler h, size_t size, memory_pool pool);
+----
+
+_Effects_: Enqueues a commands to `q` or the SYCL queue associated with `h`
+which will asynchornously allocate memory of size `size` in bytes, allocating
```

Comment:
```suggestion
which will asynchronously allocate memory of size `size` in bytes, allocating
```

Reply:
Fixed in https://github.com/intel/llvm/pull/14800/commits/2c51adf28ece21c0e91c2c0ec85440ae1153e536.